foreshadow/utils/common.py

```diff
@@ -106,3 +107,27 @@ def configure_column_sharer(self, column_sharer):
         """
         if hasattr(self, "column_sharer"):
             self.column_sharer = column_sharer
+
+
+class CustomizeParamsMixin:
+    """Mixin that customizes params returned from get_params."""
+
+    def customize_serialization_params(self, params):
+        """Select params that are only necessary.
+
+        Args:
+            params: the params returned from get_params
```

Comment:
why not make this method call get_params and force that this method only works with objects that have a working get_params method?